### PR TITLE
Restructures and adds snapshot entry point

### PR DESCRIPTION
**Pull request by thrasher-redhat**
Restructures the cmd/ folder to have a snapshot and serve subfolders. These
will have 'options' helpers to read and populate config objects from the given
yaml files.

Implements the snapshot entry point and corresponding options helper.

Minor changes to the bugzilla client.

### CODE REVIEW

**Comment by thrasher-redhat** [comment]
Bleh.  Based this off of my other branch instead of master...

Notes to self:  Don't do that.  Also wait for #3 to merge first.

File: cmd/snapshot/main.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+		configs.Sources.Database.SslMode,
+	)
+	if err != nil {
+		log.Fatalf("Error creating database client: %v", err)
+	}
+	defer dbClient.Close()
+	err = dbClient.SnapshotBugzilla(bugs)
+	if err != nil {
+		log.Fatalf("Error storing snapshot to database: %v", err)
+	}
```
Comment (DirectXMan12) [inline cmd/snapshot/main.go]: maybe write "done" at the end just to confirm?

File: cmd/snapshot/options.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+// Configs holds all of the configuration objects
+type Configs struct {
+	Sources SourceConfigs `yaml:"Sources"`
+}
+
+// populate reads the given yaml file and populates the configuration options structs
+func populate(fileName *string) (Configs, error) {
+	file, err := os.Open(*fileName)
+	if err != nil {
+		return Configs{}, fmt.Errorf("Unable to open configuration file: %v", err)
```
Comment (DirectXMan12) [inline cmd/snapshot/options.go]: errors always start with a lowercase letter.

File: pkg/db/client.go [hunk 3]
```diff
@@ -147,3 +120,30 @@ func storeBugs(tx *sql.Tx, bugs bugzilla.Bugs) error {
 
 	return nil
 }
+
+// SnapshotBugzilla removes today's bugs (if any) and stores the new bugs in a single transaction
+func (c postGresClient) SnapshotBugzilla(bugs bugzilla.Bugs) error {
```
Comment (DirectXMan12) [inline pkg/db/client.go]: o_O.  What's up with this capitalization.

**Pull request merged**